src/yunohost/certificate.py

```diff
@@ -0,0 +1,641 @@
+    statusSummaryCode = 0
+    # Critical
+    if (daysRemaining <= 0) :                                            statusSummaryCode = -30
+    # Warning, self-signed, browser will display a warning discouraging visitors to enter website
+    elif (CAtype == "Self-signed") or (CAtype == "Fake Let's Encrypt") : statusSummaryCode = -20
+    # Attention, certificate will expire soon (should be renewed automatically if Let's Encrypt)
+    elif (daysRemaining < validity_limit) :                              statusSummaryCode = -10
+    # CA not known, but still a valid certificate, so okay !
+    elif (CAtype == "Other / Unknown") :                                 statusSummaryCode =  10
+    # Let's Encrypt, great !
+    elif (CAtype == "Let's Encrypt") :                                   statusSummaryCode =  20
+
+    return { "domain"      : domain,
+             "subject"     : certSubject,
+             "CAname"      : certIssuer,
+             "CAtype"      : CAtype,
+             "validity"    : daysRemaining,
+             "summaryCode" : statusSummaryCode
+           }
+
+###############################################################################
+#   Misc small stuff ...                                                      #
+###############################################################################
+
+def _generate_account_key() :
+
+    logger.info("Generating account key ...")
+    _generate_key(account_key_file)
+    _set_permissions(account_key_file, "root", "root", 0400)
+
+def _generate_key(destinationPath) :
+
+    k = crypto.PKey()
+    k.generate_key(crypto.TYPE_RSA, key_size)
+
+    with open(destinationPath, "w") as f :
+        f.write(crypto.dump_privatekey(crypto.FILETYPE_PEM, k))
+
+def _set_permissions(path, user, group, permissions) :
+
+    uid = pwd.getpwnam(user).pw_uid
+    gid = grp.getgrnam(group).gr_gid
+
+    os.chown(path, uid, gid)
+    os.chmod(path, permissions)
+
+def _backup_current_cert(domain):
+
+    logger.info("Backuping existing certificate for domain "+domain)
+
+    cert_folder_domain = cert_folder+"/"+domain
+
+    dateTag = datetime.now().strftime("%Y%m%d.%H%M%S")
+    backup_folder = cert_folder_domain+"-backup-"+dateTag
+
+    shutil.copytree(cert_folder_domain, backup_folder)
+
+
+def _check_domain_is_correctly_configured(domain) :
+
+    public_ip = yunohost.domain.get_public_ip()
+
+    # Check if IP from DNS matches public IP
+    if not _dns_ip_match_public_ip(public_ip, domain) :
+        raise MoulinetteError(errno.EINVAL, m18n.n('certmanager_domain_dns_ip_differs_from_public_ip', domain=domain))
+
+    # Check if domain seems to be accessible through HTTP ?
+    if not _domain_is_accessible_through_HTTP(public_ip, domain) :
+        raise MoulinetteError(errno.EINVAL,  m18n.n('certmanager_domain_http_not_working', domain=domain))
+        
+def _dns_ip_match_public_ip(public_ip, domain) :
+
+    try :
+        r = requests.get("http://dns-api.org/A/"+domain)
```

Comment:
I don't know if using 'dnspython' lib is easy, but I'd like to avoid using a third party API. Note that 'dnspython' is package in Debian's `python-dns`.